src/sage/categories/simplicial_sets.py

```diff
@@ -574,6 +581,414 @@ def universal_cover(self):
+                    [           0            0            0 f2*f2inv - 1            0]
+                    [           0            0            0            0 f2*f2inv - 1]
+                    [f3*f3inv - 1            0            0            0            0]
+                    [           0 f3*f3inv - 1            0            0            0]
+                    [           0            0 f3*f3inv - 1            0            0]
+                    [           0            0            0 f3*f3inv - 1            0]
+                    [           0            0            0            0 f3*f3inv - 1]
+                    sage: Y.twisted_homology(2)
+                    Quotient module by Submodule of Ambient free module of rank 0 over the integral domain Multivariate Polynomial Ring in f2, f2inv, f3, f3inv over Integer Ring
+                    Generated by the rows of the matrix:
+                    []
+                    sage: Y.twisted_homology(1, reduced=True)
+                    Quotient module by Submodule of Ambient free module of rank 5 over the integral domain Multivariate Polynomial Ring in f2, f2inv, f3, f3inv over Integer Ring
+                    Generated by the rows of the matrix:
+                    [1 0 0 0 0]
+                    [0 1 0 0 0]
+                    [0 0 1 0 0]
+                    [0 0 0 1 0]
+                    [0 0 0 0 1]
+
+                TESTS::
+
+                    sage: X = simplicial_sets.PresentationComplex(groups.presentation.FGAbelian((3,2)))
+                    sage: TW2 = X.twisted_homology(2, reduced=True)
+                    sage: M = TW2.relations_matrix()
+                    sage: from sage.libs.singular.function import singular_function
+                    sage: vdim = singular_function("vdim")
+                    sage: vdim(M.T, ring=M.base_ring())
+                    // ** considering the image in Q[...]
+                    // ** _ is no standard basis
+                    5
+                    sage: X.universal_cover().homology(2)
+                    Z^5
+                    sage: from sage.libs.singular.function import singular_function
+
+                """
+                from sage.libs.singular.function import singular_function
+                from sage.libs.singular.option import opt_verb
+                opt_verb['not_warn_sb'] = True
+                singstd = singular_function("std")
+                singsyz = singular_function("syz")
+                singred = singular_function("reduce")
+                singlift = singular_function("lift")
+                G, d = self._universal_cover_dict()
+                phi = G.abelianization_map()
+                abelG, R, I, images = G.abelianization_to_algebra(ZZ)
+                CC = self.twisted_chain_complex()
+                M1 = CC.differential(n).T
+                M2 = CC.differential(n + 1).T
+
+                def convert_to_polynomial(p):
```

Comment:
Would it be better to define a function that convert to a polynomial and also take out positive powers of the variables?